### ISSUE TITLE
`if None is not x` cannot strip away `Optional` unlike `if x is not None`

### ISSUE THREAD

**Issue by nodakai** [python/mypy#3049]
```python
try:
    from typing import Optional  # noqa: F401
except ImportError:
    pass

def f(x):
    # type: (Optional[str]) -> None

    if x is not None:
        x.upper()
```

```
$ mypy --strict-optional /tmp/nn.py && echo OK
OK
```

```python
try:
    from typing import Optional  # noqa: F401
except ImportError:
    pass

def f(x):
    # type: (Optional[str]) -> None

    if None is not x:
        x.upper()
```

```
$ mypy --strict-optional /tmp/nn.py && echo OK
/tmp/nn.py:10: error: Some element of union has no attribute "upper"
```

```
$ mypy --version
mypy 0.501
```

- https://en.wikipedia.org/wiki/Yoda_conditions

**Comment by ambv** [comment]
This is an omission in mypy but fixing this will be low-pri. Feel free to provide a pull request.

Yoda conditions are meaningless in Python (there's no assignments in `if` statements) and are thus discouraged. They read worse for no benefit.

**Comment by pranavrajpal** [comment]
This appears to be fixed on mypy 0.910.

**Comment by JelleZijlstra** [comment]
Yes, this is fixed.
